docs/proposals/backup-restore.md

```diff
@@ -1,131 +1,261 @@
-Each artifact in the output OCI layout will be tagged with the name of source.
-For example, the `registry.k8s.io/kube-apiserver-arm64:v1.31.0` artifact will be tagged `registry.k8s.io/kube-apiserver-arm64:v1.31.0`.
+**New flags:**
 
-```bash
-% oras repo tags --oci-layout ./mirror
-registry.k8s.io/kube-apiserver-arm64:v1.31.0
-registry.k8s.io/kube-apiserver-arm64:v1.32.0
-registry.k8s.io/kube-controller-manager-arm64:v1.31.0
+- `--input <path>`: Required. Restore from a folder or archive file to registry.
+- `--exclude-referrers`: Restore the image from backup excluding referrers.
+
+**Common flags:**
+
+* `--concurrency <int>`: Number of parallel upload operations. Default: `3`.
+* `--plain-http`: Allow insecure connections to registry without SSL check.
+* `--insecure`: Allow connections to registries without valid TLS certificates.
+* `--registry-config <path>`: Path to the authentication configuration file for the registry.
+* `--username <string>`: Username for authenticating to the registry.
+* `--password <string>`: Password for authenticating to the registry.
+* `--password-stdin`: Read password from stdin.
+* `--identity-token <string>`: Use bearer token for authentication.
+* `--identity-token-stdin`: Read identity token from stdin.
+* `--ca-file <path>`: Path to custom CA certificate file.
+* `--cert-file <path>`: Path to client TLS certificate file.
+* `--key-file <path>`: Path to client private key file.
+* `--resolve <host:port:address[:address_port]>`: Customized DNS for registry.
+* `--debug`: Output debug logs (implies `--no-tty`).
+* `--distribution-spec string`: [Preview] set OCI distribution spec version and API option for target. Options: v1.1-referrers-tag, v1.1-referrers-api
+* `--no-tty`: Disable progress bars.
+
+### User Experience in the CLI
+
+The desired end-to-end user experience of using `oras backup` and `oras restore` to address the identified problems and support the outlined user scenarios is illustrated below.
+
+#### Offline Snapshot for Air-Gapped Environments
+
+Create a snapshot of a sample image `registry-a.k8s.io/kube-apiserver:v1` and its referrer (e.g. signature) for an air-gapped environment:
+
+```console
+oras backup registry-a.k8s.io/kube-apiserver:v1 --include-referrers --output airgap-snapshot.tar
 ```
 
-The backup command will also have the ability to write output to a new compressed tar file where the contents are in a single oci-layout
-format. For example:
+Upon success, the output will be:
+
+```console
+Pulled 1 manifest(s) from registry-a.k8s.io/kube-apiserver:v1
+Found 1 linked referrer(s)
+Included referrers in backup: [application/vnd.cncf.notary.signature] sha256:78833f9c870...
+Exported backup to airgap-snapshot.tar
+Backup completed: 2 manifest(s) written
+```
+
+Transfer the `.tar` file to the air-gapped system via a secured channel. Restore the tarball from local to another registry:
+
+```console
+oras restore registry-b.k8s.io/kube-apiserver:v1 --input airgap-snapshot.tar
```

Comment:
curious if we need a `--dry-run` mode or not.

Reply:
I think `--dry-run` is valuable as it enables preview before actual restore and avoid accidental overwrites in the target registry. I added as follows:

`--dry-run`: Optional. Simulate the restore process without actually uploading any artifacts to the target registry.